ouroboros-consensus-cardano/src/unstable-cardano-testlib/Test/Consensus/Cardano/ProtocolInfo.hs

```diff
@@ -307,34 +306,20 @@ mkTestProtocolInfo
               conwayProtVer                 = hfSpecProtVer Conway hardForkSpec
             , conwayMaxTxCapacityOverrides  = Mempool.mkOverrides Mempool.noOverridesMeasure
             }
-          ProtocolTransitionParamsByronToShelley {
-            transitionByronToShelleyTranslationContext = SL.toFromByronTranslationContext shelleyGenesis
-          , transitionByronToShelleyTrigger            = hfSpecTransitionTrigger Shelley hardForkSpec
-          }
-          ProtocolTransitionParamsIntraShelley {
-              transitionIntraShelleyTranslationContext = ()
-            , transitionIntraShelleyTrigger            = hfSpecTransitionTrigger Allegra hardForkSpec
-            }
-          ProtocolTransitionParamsIntraShelley {
-              transitionIntraShelleyTranslationContext = ()
-            , transitionIntraShelleyTrigger            = hfSpecTransitionTrigger Mary hardForkSpec
-            }
-          ProtocolTransitionParamsIntraShelley {
-              transitionIntraShelleyTranslationContext =
-                -- tests using this ProtocolInfo do not even reach Allegra
-                SL.exampleAlonzoGenesis
-            , transitionIntraShelleyTrigger            = hfSpecTransitionTrigger Alonzo hardForkSpec
-            }
-          ProtocolTransitionParamsIntraShelley {
-              transitionIntraShelleyTranslationContext = ()
-            , transitionIntraShelleyTrigger            = hfSpecTransitionTrigger Babbage hardForkSpec
-            }
-          ProtocolTransitionParamsIntraShelley {
-              transitionIntraShelleyTranslationContext =
-                -- tests using this ProtocolInfo do not even reach Allegra
-                SL.exampleConwayGenesis
-            , transitionIntraShelleyTrigger            = hfSpecTransitionTrigger Conway hardForkSpec
+          CardanoHardForkTriggers' {
+              triggerHardForkShelley = hfSpecTransitionTrigger Shelley hardForkSpec
+            , triggerHardForkAllegra = hfSpecTransitionTrigger Allegra hardForkSpec
+            , triggerHardForkMary    = hfSpecTransitionTrigger Mary    hardForkSpec
+            , triggerHardForkAlonzo  = hfSpecTransitionTrigger Alonzo  hardForkSpec
+            , triggerHardForkBabbage = hfSpecTransitionTrigger Babbage hardForkSpec
+            , triggerHardForkConway  = hfSpecTransitionTrigger Conway  hardForkSpec
             }
+          ( L.mkLatestTransitionConfig
+              shelleyGenesis
+              -- tests using this ProtocolInfo do not even reach Allegra
```

Comment:
If why know why these tests do not reach `Allegra` perhaps we can explain why.

Reply:
Oh thanks for making me double-check that; you did originally factor out this module in #225, but the comment was added in 3df408dc1435b584e72d2565e2fdfd9417172536 (#283), but I think it is incorrect as your [Cardano LocalTxSubmissionServer test](https://github.com/input-output-hk/ouroboros-consensus/blob/a70f0f35594923dc2412edcc76a03bb92c50583c/ouroboros-consensus-cardano/test/cardano-test/Test/Consensus/Cardano/MiniProtocol/LocalTxSubmission/Server.hs) actually go all the way to Conway (in contrast to the ThreadNet test, which indeed never fork into Alonzo). Will correct this comment and make it more precise :+1:

Reply:
9751e958013a996b7f346a60947ffbf7b9f3827a